### DISCUSSION TITLE
FileNotFound Error Help

### DISCUSSION THREAD

**Discussion by roman-tasi** [jertel/elastalert2#635] [Q&A] [answered]
I am getting this error in my docker logs:

`FileNotFoundError: [Errno 2] No such file or directory: '/etc/elastalert/smtp_auth_user.yaml'`

However that file exists in that path so I'm confused. Also I gave it 777 permissions but still have the same error. Can anyone help?

This is the field referenced in my `rules.yaml` :

`smtp_auth_file: "/etc/elastalert/smtp_auth_user.yaml"`

**Comment by jertel** [comment, marked as answer]
Are you using Docker? If so, have you mapped that file into the Docker container using the `-v` arg?

**Reply by roman-tasi** [reply]
I am using Docker. This is the command I am currently running:

`docker run -d --name elastalert --restart=always -v $(pwd)/elastalert.yaml:/opt/elastalert/config.yaml -v $(pwd)/elastalert_rules:/opt/elastalert/rules ghcr.io/jertel/elastalert2/elastalert2 --verbose`

Do I need to add another `-v` arg? And if I do, what should it look like?

**Reply by jertel** [reply]
```
-v /etc/elastalert/smtp_auth_user.yaml:/etc/elastalert/smtp_auth_user.yaml
```
Out of respect for those supporting this project and per #11, please review Docker documentation and tutorials so that you're in a better position to find answers like this on your own.

**Reply by roman-tasi** [reply]
Thank you... however I am getting the same `FileNotFoundError` even with the new `-v` argument. This is the command I am running:

```
docker run -d --name elastalert --restart=always \
  -v $(pwd)/elastalert.yaml:/opt/elastalert/config.yaml \
  -v $(pwd)/elastalert_rules:/opt/elastalert/rules \
  -v $(pwd)/smtp_auth_user.yaml:/opt/elastalert/smtp_auth_user.yaml \
  ghcr.io/jertel/elastalert2/elastalert2 --verbose
```

Any idea how I can fix this?

**Reply by jertel** [reply]
Includes logs and configs and we'll try to assist.

**Reply by jertel** [reply]
Your `-v` param is incorrect. The right half of the `:` needs to point to the same SMTP file location that you specified in your rule config. Again, please read the Docker instructions to understand how file mapping works.
